Project/Src/Document/HighlightingStrategy/HighlightingManager.cs

```diff
@@ -132,10 +133,10 @@ public IHighlightingStrategy FindHighlighter(string name)
 
         public IHighlightingStrategy FindHighlighterForFile(string fileName)
         {
-            var extension = Path.GetExtension(fileName).ToUpperInvariant();
-            if (extensionsToName.TryGetValue(extension, out var highlighterName))
-                return FindHighlighter(highlighterName);
-            return DefaultHighlighting;
+            var filenameUpper = fileName.ToUpperInvariant();
+
+            var highlighterName = extensionsToName.FirstOrDefault(e => filenameUpper.EndsWith(e.Key));
```

Comment:
Would this work?
```suggestion
            var highlighterName = extensionsToName.FirstOrDefault(e => filenameUpper.EndsWith(e.Key, StringComparison.OrdinalIgnoreCase));
```


It would be interesting to run a simple benchmark on this and see what one is more optimal

Reply:
done.